satpy/dependency_tree.py

```diff
@@ -424,6 +430,7 @@ def _find_compositor(self, dataset_key, query):
         # one or more modifications if it has modifiers see if we can find
         # the unmodified version first
 
+        orig_query = dataset_key
```

Comment:
1. are we sure this is a query, not an id?
2. should we rename the function argument instead?

Reply:
I'm honestly not sure, but I think so. This is reminding me that I wanted to add more type annotations to this stuff for this reason, but forgot after needing some last minute changes in other parts of the code. If I remember correctly the information passed by the user gets turned into a DataQuery in one of the top-level tree methods and gets passed around from there.